ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master__cnv2.5_4.6.yaml

```diff
@@ -0,0 +1,156 @@
+base_images:
+  base:
+    name: "4.6"
+    namespace: ocp
+    tag: base
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.13
+canonical_go_repository: kubevirt.io/kubevirt
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tag_specification:
+  name: "4.6"
+  namespace: ocp
+tests:
+- artifact_dir: /tmp/artifacts
+  as: e2e
+  cron: 2 3 * * *
+  steps:
+    cluster_profile: azure4
+    test:
+    - as: test
+      cli: latest
+      commands: |
+        set -ex
+
+        # disable default catalog source
+        oc patch OperatorHub cluster --type json -p '[{"op": "add", "path": "/spec/disableAllDefaultSources", "value": true}]'
+
+        authfile=/tmp/authfile
+        # get authfile from cluster
+        oc get secret/pull-secret -n openshift-config -o json | jq -r '.data.".dockerconfigjson"' | base64 -d > "$authfile"
+
+        # login into brew image registry
+        docker login --username "$BREW_IMAGE_REGISTRY_USERNAME" --password-stdin brew.registry.redhat.io < $BREW_IMAGE_REGISTRY_TOKEN_PATH
+
+        # update authfile with credentials from docker config
+        jq -c '.auths + '"$(jq '{"brew.registry.redhat.io": .auths."brew.registry.redhat.io" }' $HOME/.docker/config.json)"' | {"auths": .}' "$authfile" > "${authfile}.new"
+
+        # update cluster secret
+        oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson="${authfile}.new"
+
+        # install imageContentSourcePolicy
+        oc create -f - << EOF
+        apiVersion: operator.openshift.io/v1alpha1
+        kind: ImageContentSourcePolicy
+        metadata:
+          name: brew-registry
+        spec:
+          repositoryDigestMirrors:
+          - mirrors:
+            - brew.registry.redhat.io
+            source: registry.redhat.io
+          - mirrors:
+            - brew.registry.redhat.io
+            source: registry.stage.redhat.io
+          - mirrors:
+            - brew.registry.redhat.io
+            source: registry-proxy.engineering.redhat.com
+        EOF
+
+        # wait for machineconfigpool to get updated
+        oc wait mcp/master --for condition=updated --timeout=600s
+        oc wait mcp/worker --for condition=updated --timeout=600s
+
+        # install index image
+        oc create -f - << EOF
+        apiVersion: operators.coreos.com/v1alpha1
+        kind: CatalogSource
+        metadata:
+          name: brew-catalog-source
+          namespace: openshift-marketplace
+        spec:
+          sourceType: grpc
+          image: brew.registry.redhat.io/rh-osbs/iib-pub-pending:v$CNV_VERSION
+          displayName: Brew Catalog Source
+          publisher: grpc
+        EOF
+
+        # deploy operator
+        oc create -f - << EOF
```

Comment:
@orenc1 I tested the script locally with crc, it's complaining that it doesn't know the type. I was just guessing it could be this one to deploy the cnv operator, and thinking that the mco is not working on crc. Am I correct, or what do I need to change?

Reply:
@dhiller, you don't have the `HyperConverged` type at this point because the HCO CRD hasn't been deployed yet. It should be deployed by the OLM upon creating a valid subscription, which in turn creates the install plan.
Before creating the HCO CR, you should create the subscription object:
```
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: kubevirt-hyperconverged
  namespace: openshift-cnv
  labels:
    operators.coreos.com/kubevirt-hyperconverged.openshift-cnv: ''
spec:
  channel: stable
  installPlanApproval: Automatic
  name: kubevirt-hyperconverged
  source: brew-catalog-source
  sourceNamespace: openshift-marketplace
  startingCSV: kubevirt-hyperconverged-operator.v2.5.1
```
Also, you are correct that MCO is not operating on single-node clusters (like CRC), so you can test it using a PSI or QuickLab cluster.

Reply:
OK, thanks for the feedback! Will test this on a quicklab cluster now. Do I need to wait somehow after creating the subscription? And what else do I need to do from your POV?

Reply:
After creating the subscription and everything was successfully resolved, you should expect a creation of CSV and InstallPlan objects in the installation namespace (`openshift-cnv`) and all CNV CRDs, as well as deployments of the operators being deployed by OLM, and eventually the pods themselves. This can take up to several minutes.
At the time the CRDs are deployed you can create the HCO CR to complete the installation.